core/src/main/java/com/microsoft/applicationinsights/internal/shutdown/SDKShutdownActivity.java

```diff
@@ -77,6 +77,20 @@ public void stop(long timeout, TimeUnit timeUnit) {
             });
         }
 
+        private volatile long perThreadTimeout = 10L;
+        private volatile TimeUnit perThreadTimeUnit = TimeUnit.SECONDS;
+
+        public synchronized long getPerThreadTimeout() {
```

Comment:
Why do you need synchronized here? As far as run method is concerned it is still synchronized and will get consistent values. 

I understand that threadB can still call async getPerThreadTimeOut() in a scenario but that should be fine as far as i can think of because set method will still be synchronzied and hence the values won't change without blocking.

Reply:
I think I intended to make this configurable, but didn't for some reason. The setter below is also synchronized, but I suppose this doesn't matter that much.

I can remove `synchronized` if needed, i.e. if we don't need this to be configurable.

Reply:
I don't think atm we need this configurable. Would be good to remove for now and bring it back when needed.